### PR TITLE
feat/4713 Restrict online edition file extensions

### PR DESCRIPTION
**Pull request by grignards**
See #4713

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

***Testing instructions***

Collabora setup:
- `ip addr show`, note the IP address of the `docker0` interface
- edit `backend/docker-compose-optional.yml` to set `domain` and `server_name` to the noted IP (with the format as described in the file)
- run collabora with `docker-compose -f docker-compose-optional.yml` up -d`

Backend setup:

- add `collaborative_document_edition` to `app.enabled` in `development.ini`
- `collaborative_document_edition.software = collabora` in `development.ini`
- `collaborative_document_edition.collabora.base_url = http://<NOTED_IP_ADDRESS>:9980` in `development.ini`
- run the backend with `run_dev_backend.sh`

You can then play with the value of `collaborative_document_edition.enabled_extensions` to restrict the file extensions available in collabora.

### CODE REVIEW

File: backend/tracim_backend/applications/collaborative_document_edition/lib.py [hunk 1]
```diff
@@ -69,19 +81,27 @@ def _get_template_list(self) -> typing.List[str]:
...
+            if isfile(join(self._config.COLLABORATIVE_DOCUMENT_EDITION__FILE_TEMPLATE_DIR, entry))
+        ]
+
+        if not self._config.COLLABORATIVE_DOCUMENT_EDITION__ENABLED_EXTENSIONS:
+            return template_filenames
+
+        return [
+            filename
+            for filename in template_filenames
+            if os.path.splitext(filename)[1][1:]
```
Comment (inkhey) [inline backend/tracim_backend/applications/collaborative_document_edition/lib.py:102]: What happen if one file has no extension here ?

Comment (grignards) [reply backend/tracim_backend/applications/collaborative_document_edition/lib.py:102]: `splitext()[1]` would return an empty string, so the file wouldn't be taken into account, which is logical I think.

Comment (inkhey) [reply backend/tracim_backend/applications/collaborative_document_edition/lib.py:102]: ok.

**Review by inkhey — APPROVED** [review]

**Comment by MariePrandstetter** [comment]
tested on branch

**Pull request merged**